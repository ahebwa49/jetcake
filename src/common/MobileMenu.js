import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { withRouter } from "react-router-dom";
import { addUser } from "../actions/addUser";
import MenuButtonClose from "./MenuButtonClose";

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

const mapDispatchToProps = dispatch => ({
  addNewUser: user => {
    dispatch(addUser(user));
  }
});

class MobileMenu extends Component {
  constructor(props) {
    super(props);
    this.handleLogout = this.handleLogout.bind(this);
  }

  handleLogout() {
    fetch("http://localhost:4000/logout", {
      method: "GET",
      credentials: "include"
    })
      .then(response => {
        return response.json();
      })
      .then(async data => {
        await this.props.addNewUser(data);
        this.props.history.push("/signin");
      });
  }
  render() {
    return (
      <div className="mobileMenu">
        <div className="listItems">
          {this.props.user.username || this.props.user.email ? (
            <>
              <div>{this.props.user.username || this.props.user.email}</div>
              <div>
                <Link to="/profile">Profile</Link>
              </div>
              <div onClick={this.handleLogout}>logout</div>
            </>
          ) : (
            <>
              <div>
                <Link to="/signin">Sign in</Link>
              </div>
              <div>
                {" "}
                <Link to="/signup">Register</Link>
              </div>
            </>
          )}
        </div>
        <div className="close">
          <MenuButtonClose handleSidebarHide={this.props.handleSidebarHide} />
        </div>
      </div>
    );
  }
}
export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(MobileMenu)
);
