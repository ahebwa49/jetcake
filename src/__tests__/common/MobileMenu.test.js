import React from "react";
import configureStore from "redux-mock-store";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import toJson from "enzyme-to-json";

import MobileMenu from "../../common/MobileMenu";

const mockStore = configureStore([]);

let component;

component = shallow(<MobileMenu />);

test("MobileMenu component renders without crashing", () => {
  expect(toJson(component)).toMatchSnapshot();
});
