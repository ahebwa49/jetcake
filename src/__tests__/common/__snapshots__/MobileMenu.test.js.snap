// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu component renders without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
